### DISCUSSION TITLE
Where do I find JXRDecApp.exe for JXR images?

### DISCUSSION THREAD

**Discussion by Sawtaytoes** [ImageMagick/ImageMagick#5651] [Help]
I downloaded `ImageMagick-7.1.0-portable-Q16-HDRI-x64` from the website and noticed there's no `JXRDecApp.exe` for decoding JXR used in the conversion process nor `JXREncApp.exe` for encoding.

This is the error I get when running `magick image.jxr image.exr`:
```sh
'"JXRDecApp.exe"' is not recognized as an internal or external command,
operable program or batch file.
The system cannot find the file specified.
magick: delegate failed `cmd.exe /c (move "%i" "%i.jxr" >nul) & ("JXRDecApp.exe" -i "%i.jxr" -o "%o.pnm") & (move "%i.jxr" "%i" >nul) & (move "%o.pnm" "%o" >nul)' @ error/delegate.c/InvokeDelegate/1911.
magick: unable to open file 'C:/Users/USERNAME/AppData/Local/Temp/magick-UY9hPjNtrYnRjnLtkirKcYpOJ4IBfyq4': No such file or directory @ error/constitute.c/ReadImage/781.
```

My goal is to convert JXR images to other HDR formats like **OpenEXR**, **HEIF**, and **HEIC**.

**Comment by fmw42** [comment]
It is not a bug. Not every possible image format is included in the portable release.  I suppose you could ask that it be included in the future by making an Enhancement request.

**Reply by Sawtaytoes** [reply]
I'm wondering if the portable release is missing content that I could otherwise acquire from another bundle on the website.

Is this part of the installed or any other bundle? If not, then I will make an enhancement request.

**Reply by dlemstra** [reply]
You will need to compile that yourself. I found source code for that application here: https://github.com/curasystems/jxrlib. We will not make this part of our releases.

**Reply by MajorMurphy** [reply]
FYI for anyone still searching for this, a more recently updated fork is available here: https://github.com/4creators/jxrlib

**Reply by friedhumanpie** [reply]
And an FYI for anyone who has come down this route to convert to JXR for HDR content within Windows - [jxrlib](https://github.com/4creators/jxrlib) does not seem to retain image information above 8 bit colour when converting to it from tif or bmp. If your intent is conversion from JXR however it will function as expected.

**Comment by almighty-loaf** [comment]
Pre-compiled binaries for Windows: https://github.com/Knewest/precompiled-jxrlib-binaries/releases
